test/socket_constructor_test.cpp

```diff
@@ -13,6 +13,9 @@ TEST(Socket, ConstructorIPv4)
 
 	EXPECT_ANY_THROW(Socket s(Domain::IPv4, Type::TCP, 41));
 	EXPECT_ANY_THROW(Socket s(Domain::IPv4, Type::UDP, 6));
+
+	EXPECT_NO_THROW(Socket otherS(Socket(Domain::IPv4, Type::TCP)));
```

Comment:
@tkhurana96 try with a `vector.emplace()` kind of method then, which is more practical example.